### ISSUE TITLE
Missing default value in schema output for arrays

### ISSUE THREAD

**Issue by KeweiCodes** [cockroachdb/activerecord-cockroachdb-adapter#136]
We have this migration in our project:

```ruby    
  add_column :users, :google_ids, :text, array: true, default: []
```

However after running db:migrate, the schema.rb generated is:

```ruby
  t.text "google_ids", array: true
```

Notice the missing default value. 
This is a problem if we want to do recreate our cockroach DB from the schema.

Any ideas on how this can be solved? Thanks!

**Comment by rafiss** [comment]
@KeweiCodes Thanks for the report! Can you send some more information?

- What version of CockroachDB are you using?
- What is the result of this query?
```
              SELECT a.attname, format_type(a.atttypid, a.atttypmod),
                     pg_get_expr(d.adbin, d.adrelid), a.attnotnull, a.atttypid, a.atttypmod,
                     c.collname, col_description(a.attrelid, a.attnum) AS comment
                FROM pg_attribute a
                LEFT JOIN pg_attrdef d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
                LEFT JOIN pg_type t ON a.atttypid = t.oid
                LEFT JOIN pg_collation c ON a.attcollation = c.oid AND a.attcollation <> t.typcollation
               WHERE a.attrelid = 'users'::regclass
                 AND a.attnum > 0 AND NOT a.attisdropped
               ORDER BY a.attnum;
```

**Comment by KeweiCodes** [comment]
Hi @rafiss we are using v20.1.4
See below for the result of the query, the column in question named `google_shortcut_ids`
<img width="939" alt="Screen Shot 2020-10-06 at 8 34 48 AM" src="https://user-images.githubusercontent.com/10202555/95123805-033e0700-07af-11eb-97b4-761c03388c30.png">

**Comment by rafiss** [comment]
Thanks for the info. I believe the problem happening here is this: https://github.com/cockroachdb/cockroach/issues/55320 (I just filed this issue). I'll work on getting a fix prioritized.

**Comment by rafiss** [comment]
I spent some time working on a quick way to address this issue with changes to the adapter. I was trying to make the adapter understand the ARRAY syntax returned by CockroachDB by modifying this function: https://github.com/cockroachdb/activerecord-cockroachdb-adapter/blob/f9b0d4e8f494d54672db3c914d50d866dce3762a/lib/active_record/connection_adapters/cockroachdb_adapter.rb#L220-L224

However, I don't think there's a simple way. It's easy enough to write a regex that can detect the `ARRAY[...]` syntax, but it is not easy to handle some edge cases for the array contents, without writing a lot of parsing logic.

Consider the following inputs and desired outputs:
```
ARRAY['a'::string, 'b'::string]  ->  {a,b}
ARRAY['a'::string, 'b,c'::string]  ->  {a,"b,c"}
ARRAY['a'::string, ''::string]  ->  {a,""}
ARRAY[e'a\'::string'::STRING, e'\'::string'::STRING]  ->  {"a'::string","'::string"}
ARRAY[e'a\'::string, \'::string'::STRING]  ->  {"a''::string, ''::string"}
```

**Comment by rafiss** [comment]
I will close this since https://github.com/cockroachdb/activerecord-cockroachdb-adapter/pull/144 handles the common case. Any other cases will be addressed in CockroachDB 21.1.0.
